### PR TITLE
LDAP authentication

### PR DESCRIPTION
**Pull request by Haarolean**

### CODE REVIEW

**Comment by DementevNikita** [comment]
@Haarolean, please define corresponded entries in application.yaml and update auth.type comment. Also, we need to test this using our demo stand somehow, it'd be great if you dug into that.

File: docker/docker/examples/ldap.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      - database
+      - ldap
+    ports:
+      - 8080:8080
+
+  ldap:
+    image: rroemhild/test-openldap:latest
+    hostname: "ldap"
+
+  prometheus:
```
Comment (DementevNikita) [inline docker/docker/examples/ldap.yml:34]: Neither of prometheus and alertmanager are necessary here

Comment (Haarolean) [reply docker/docker/examples/ldap.yml]: done

File: odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/config/LDAPSecurityConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import org.springframework.security.config.web.server.ServerHttpSecurity;
+import org.springframework.security.ldap.authentication.BindAuthenticator;
+import org.springframework.security.ldap.authentication.LdapAuthenticationProvider;
+import org.springframework.security.web.server.SecurityWebFilterChain;
+import org.springframework.security.web.server.util.matcher.PathPatternParserServerWebExchangeMatcher;
+
+@Configuration
+@EnableWebFluxSecurity
+@ConditionalOnProperty(value = "auth.type", havingValue = "LDAP")
+@Log4j2
```
Comment (damirabdul) [inline odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/config/LDAPSecurityConfiguration.java:27]: why do you need it?

Comment (Haarolean) [reply odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/config/LDAPSecurityConfiguration.java:27]: Logging on line 58

Comment (damirabdul) [reply odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/config/LDAPSecurityConfiguration.java:27]: yes, but we are using logback in project via @Slf4j facade

Comment (Haarolean) [reply odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/config/LDAPSecurityConfiguration.java:27]: Oopsie, copypasted from some other place.

**Review by DementevNikita — APPROVED** [review]

**Pull request merged**